app/src/main/java/com/mandysoftware/livrenouveau/SearchActivity.kt

```diff
@@ -0,0 +1,274 @@
+package com.mandysoftware.livrenouveau
+
+import android.app.Activity
+import android.content.Intent
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.SearchView
+import android.widget.TextView
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.content.ContextCompat
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.mandysoftware.wordutil.CompositeDictionary
+import java.io.File
+
+class SearchActivity : AppCompatActivity(), CompositeDictionary.DictionaryCallback {
+
+    val TAG = "SearchActivity"
+
+    private lateinit var searchView: SearchView
+    private lateinit var rvSuggestions: RecyclerView
+
+    private val definitionActivityResultLauncher = registerForActivityResult(
+        ActivityResultContracts.StartActivityForResult()) { result ->
+        if (result.resultCode == Activity.RESULT_OK) {
+            val data = result.data
+            if (data != null) {
+                val word = data.getStringExtra("word")
+                // do something with the returned word
+                runOnUiThread {
+
+                    searchView.setQuery(word, false) //false because we don't want auto search
+                    searchView.requestFocus() // focus for simplicity
+                }
+            }
+        }
+    }
+
+    var dictionary : CompositeDictionary? = null
+
+    inner class SuggestionsAdapter(private val suggestions: List<String>) :
+        RecyclerView.Adapter<SuggestionsAdapter.ViewHolder>() {
```

Comment:
For readability, let's place the `{` here onto the next line.

Align this `{` with the `i` in `inner` (i.e., 4 spaces between the start of the line and `{`).